mesonbuild/mparser.py

```diff
@@ -113,6 +114,8 @@ def __init__(self, code: str):
                          'endif', 'and', 'or', 'not', 'foreach', 'endforeach',
                          'in', 'continue', 'break'}
         self.future_keywords = {'return'}
+        if 'MESON_UNIT_TEST' in os.environ:
+            self.keywords.update({'with', 'endwith'})
```

Comment:
I'm wondering if we should already add `with` and `endwith` into `self.future_keywords` because this kind of statement could be useful in the future, so warning for future keyword could be useful already?